core/src/main/scala/kafka/consumer/ZookeeperConsumerConnector.scala

```diff
@@ -588,7 +588,10 @@ private[kafka] class ZookeeperConsumerConnector(val config: ConsumerConfig,
             if (doRebalance)
               syncedRebalance
           } catch {
-            case t: Throwable => error("error during syncedRebalance", t)
+            case t: Throwable => {
+              error("error during syncedRebalance", t)
+              shutdown()
```

Comment:
This is a somewhat significant change in the old consumer. Previously the consumer would actually stay alive after a series of rebalance failures and could succeed on a successful subsequent rebalance trigger. So for e.g., with this change one consumer that has high latency issues with zookeeper and delay in releasing partition ownership could cause all the other (normal) consumers to shutdown.

Even in the new consumer a consumer that is problematic within the group is not shutdown. It is marked as failed but it can definitely rejoin the group.